### PR TITLE
Add acceptance tests

### PR DESCRIPTION
**Pull request by mat1010**
Hi,

this PR adds litmus acceptance tests on

- ubuntu 14.04, 16.04, 18.04 and 20.04
- debian 8, 9, 10
- centos 6, 7, 8
- scientific 6, 7
- oracle 6, 7

for Puppet 5 + 6

Unfortunately we can't test with Puppet 4 using litmus - so this change would also drop the unit tests for Puppet 4. Due to the fact htat Puppet 4 is already end of life I really wouldn't invest much time here for creating tests. What do you think?

### CODE REVIEW

File: spec/acceptance/sogo_spec.rb [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+          file_line { 'fix_sogo_systemd_unit':
+            ensure => 'absent',
+            path => '/usr/lib/systemd/system/sogod.service',
+            line => 'PIDFile=/var/run/sogo/sogo.pid',
+            notify => Service['sogod'],
+          }
+	}
+      }
+      'Debian': {
+	$extra_packages = ['sope4.9-gdl1-postgresql']
```
Comment (Thor77) [inline spec/acceptance/sogo_spec.rb:20]: can you indent this with two spaces as well, please?

File: spec/acceptance/sogo_spec.rb [hunk 2]
```diff
@@ -0,0 +1,88 @@
+require 'spec_helper_acceptance'
+
+describe '::sogo' do
+  pp = <<-PUPPETCODE
```
Comment (Thor77) [inline spec/acceptance/sogo_spec.rb:4]: Is there an option to save the state-code into a separate file instead of inlining it?

**Review by Thor77 — CHANGES_REQUESTED** [review]
Thanks for the PR! I left two comments, please take a look.

**Comment by Thor77** [comment]
> Unfortunately we can't test with Puppet 4 using litmus - so this change would also drop the unit tests for Puppet 4. Due to the fact htat Puppet 4 is already end of life I really wouldn't invest much time here for creating tests. What do you think?

I agree, tests for Puppet 4 are not really required

File: spec/acceptance/sogo_spec.rb (hunk 2 above)
Comment (mat1010) [reply spec/acceptance/sogo_spec.rb:4]: I think we could do that by declaring the variable, which contains the manifest, in another ruby file and require it, like the `spec_helper_acceptance` file within the `sogo_spec.rb`.
There seems to be no native rspec / litmus way to do that. What would be your goal to achieve? Less complexity within that single file? I also checked some other possibilities and how this is being handled in other modules. It seems to be the way [promoted by puppet](https://puppetlabs.github.io/litmus/Litmus-test-examples.html), which is also used in some popular, official, modules like [apache](/blob/a51b6a6452e9af44a21158ebc3de4340da9a431c/spec/acceptance/class_spec.rb#L40), [motd](https://github.com/puppetlabs/puppetlabs-motd/blob/cc231ef6228172d82a7c346eb9022804954cdbbc/spec/acceptance/motd_spec.rb#L11), [stdlib](https://github.com/puppetlabs/puppetlabs-stdlib/blob/13eb80c918a4b51429b4d707385c6c941b3bef17/spec/acceptance/anchor_spec.rb#L5) and many others.

We could also switch to a slightly different syntax and do it likes this
```
let(:pp) do
  <<-PUPPETCODE
    $database = 'postgresql://sogo@127.0.0.1/sogo'

    case $facts['os']['family'] {
      'RedHat': {
        $extra_packages = ['sope49-gdl1-postgresql']
        if versioncmp($facts['os']['release']['major'], '7') >= 0 {
          file_line { 'fix_sogo_systemd_unit':
            ensure => 'absent',
            path => '/usr/lib/systemd/system/sogod.service',
            line => 'PIDFile=/var/run/sogo/sogo.pid',
            notify => Service['sogod'],
          }
        }
      }
    ...
  PUPPETCODE
end
``` 

But we still would have it within the same file.

File: spec/acceptance/sogo_spec.rb (hunk 1 above)
Comment (mat1010) [reply spec/acceptance/sogo_spec.rb:20]: Fixed, a lost tab made it look good within vi until I fixed my trailing whitespace highlighter :-D

File: spec/acceptance/sogo_spec.rb (hunk 2 above)
Comment (Thor77) [reply spec/acceptance/sogo_spec.rb:4]: Hm, yeah, I was hoping for a native litmus way, but it seems like there's none :/
Yeah, less complexity/better overview would have been my goal.

**Review by Thor77 — APPROVED** [review]

**Pull request merged**